test/jdk/java/lang/Class/getEnclosingClass/EnclosingClassTest.java

```diff
@@ -59,71 +48,172 @@
  * e o x x x
  */
 
-import java.util.List;
-import java.util.LinkedList;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
 import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 import common.TestMe;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/*
+* @test
+* @bug 4992173 4992170
+* @library /test/lib
+* @build jdk.test.lib.process.* EnclosingClassTest
+* @run testng/othervm EnclosingClassTest
+* @summary Check getEnclosingClass and other methods
+* @author Peter von der Ah\u00e9
+*/
+
 
 public class EnclosingClassTest {
-    static void info(Class<?> c, Class<?> encClass, String desc) {
-        if (!"".equals(desc))
+    private static final String SRC_DIR = System.getProperty("test.src");
+    private static final String ENCLOSING_CLASS_SRC = SRC_DIR + "/EnclosingClass.java";
+
+    @BeforeClass
+    public void createEnclosingClasses() throws Throwable {
+        Path enclosingPath = Paths.get(ENCLOSING_CLASS_SRC);
+        Path pkg1Dir = Paths.get(SRC_DIR + "/pkg1");
+        Path pkg2Dir = Paths.get(SRC_DIR + "/pkg1/pkg2");
+        Path pkg1File = Paths.get(SRC_DIR + "/pkg1/EnclosingClass.java");
+        Path pkg2File = Paths.get(SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
+        try {
+            Files.deleteIfExists(pkg2File);
+            Files.deleteIfExists(pkg2Dir);
+            Files.deleteIfExists(pkg1File);
+            Files.deleteIfExists(pkg1Dir);
+            Files.createDirectories(pkg2Dir);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        createAndWriteEnclosingClasses(enclosingPath, pkg1File, "pkg1");
+        createAndWriteEnclosingClasses(enclosingPath, pkg2File, "pkg1.pkg2");
+
+        String javacPath = JDKToolFinder.getJDKTool("javac");
```

Comment:
You can use `jdk.test.lib.compiler.CompilerUtils` test library to compile the file in process.

Reply:
I tried to use the CopmilerUtils to compile the file, But this utility is not able to  the dependencies of the class. Example it is not able to find the common.TestMe class while try to compile the EnclosingClass.java.

Reply:
You need to add `--source-path` where it can find the dependent source files.